### PR TITLE
20_2 DataGrid onRowUpdating: docissues (cp)

### PR DESCRIPTION
**Pull request by pavel-2020**
Original PR: https://github.com/DevExpress/devextreme-documentation/pull/1891

### CODE REVIEW

**Review by RomanTsukanov — APPROVED** [review]

File: includes/datagrid-ref-confirm-action-and-validate-data.md [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+            ${{apiMember}}: function(e) {
+                const deferred = $.Deferred();
+                const promptPromise = DevExpress.ui.dialog.confirm("Are you sure?", "Confirm changes");
+                promptPromise.done((dialogResult) => {
+                    if (dialogResult) {
+                        let params = "";
+                        for (let key in e.newData) {
+                            params += `${key}=${e.newData[key]}&`;
+                        }
+                        params = params.slice(0, -1);
```
Comment (jtoming830) [inline includes/datagrid-ref-confirm-action-and-validate-data.md]: to delete last '&'?

Comment (jtoming830) [reply includes/datagrid-ref-confirm-action-and-validate-data.md]: Why not approach from Custom Data Source demo?

Comment (pavel-2020) [reply includes/datagrid-ref-confirm-action-and-validate-data.md]: Yes.

Comment (jtoming830) [reply includes/datagrid-ref-confirm-action-and-validate-data.md]: You can pass e.newData as data to the $.ajax - it should convert object to string. 
https://api.jquery.com/jquery.ajax/

File: includes/datagrid-ref-confirm-action-and-validate-data.md [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+            async ${{functionName}}(e) {
+                const isCanceled = async () => {
+                    const dialogResult = await dialog.confirm("Are you sure?", "Confirm changes");
+                    if (dialogResult) {
+                        let params = "?";
+                        for (let key in e.newData) {
+                            params += `${key}=${e.newData[key]}&`;
+                        }
+                        params = params.slice(0, -1);
+                        const validationResult = await fetch(`https://url/to/your/validation/service${params}`);
```
Comment (jtoming830) [inline includes/datagrid-ref-confirm-action-and-validate-data.md:121]: why not pass params as body to fetch?
https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch

**Pull request merged**